### PR TITLE
Adding Employee / Category Mappings

### PR DESCRIPTION
**Pull request by Shwetabhk**
Adding new tables for employee and category mappings

### CODE REVIEW

**Comment by Shwetabhk** [comment]
Payload of API
```json
{
    "source_employee": {
        "id": null
    },
    "destination_employee": {
        "id": null
    },
    "destination_vendor": {
        "id": null
    },
    "destination_card_account": {
        "id": null
    },
    "workspace": null
}
```

**Comment by Shwetabhk** [comment]
Get Response of API `GET api/workspaces/2/mappings/employee/`
```
{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 1,
            "source_employee": {
                "attribute_type": "CATEGORY",
                "display_name": "Category",
                "value": "00",
                "source_id": "110351",
                "auto_mapped": false,
                "auto_created": false,
                "active": null,
                "detail": null,
                "created_at": "2021-04-12T14:53:34.572984Z",
                "updated_at": "2021-04-12T14:53:34.573030Z",
                "workspace": 4
            },
            "destination_employee": {
                "attribute_type": "CUSTOMER",
                "display_name": "customer",
                "value": "Ashwinnnnnn",
                "destination_id": "103",
                "auto_created": false,
                "active": null,
                "detail": null,
                "created_at": "2021-04-13T10:19:48.710815Z",
                "updated_at": "2021-04-13T10:19:48.710878Z",
                "workspace": 1
            },
            "destination_vendor": {
                "attribute_type": "CUSTOMER",
                "display_name": "customer",
                "value": "Ashwinnnnnn",
                "destination_id": "103",
                "auto_created": false,
                "active": null,
                "detail": null,
                "created_at": "2021-04-13T10:19:48.710815Z",
                "updated_at": "2021-04-13T10:19:48.710878Z",
                "workspace": 1
            },
            "destination_card_account": {
                "attribute_type": "CUSTOMER",
                "display_name": "customer",
                "value": "Ashwinnnnnn",
                "destination_id": "103",
                "auto_created": false,
                "active": null,
                "detail": null,
                "created_at": "2021-04-13T10:19:48.710815Z",
                "updated_at": "2021-04-13T10:19:48.710878Z",
                "workspace": 1
            },
            "created_at": "2021-07-23T09:54:10.252670Z",
            "updated_at": "2021-07-23T09:54:10.252913Z",
            "workspace": 2
        }
    ]
}
```

**Comment by Shwetabhk** [comment]
@ashwin1111 You can carry it forward from here

File: fyle_accounting_mappings/views.py [hunk 1]
```diff
@@ -108,3 +108,15 @@ def post(self, request, *args, **kwargs):
...
                 }
             )
+
+
+class EmployeeMappingsView(ListCreateAPIView):
+    """
+    Employee Mappings View
+    """
+    authentication_classes = []
+    permission_classes = []
```
Comment (ashwin1111) [inline fyle_accounting_mappings/views.py:118]: TODO: remove

File: fyle_accounting_mappings/models.py [hunk 2]
```diff
@@ -538,3 +538,67 @@ def auto_map_ccc_employees(destination_type: str, default_ccc_account_id: str, w
...
+            source_employee_id=source_employee_id,
+            workspace=workspace,
+            defaults={
+                'destination_employee_id': destination_employee_id,
+                'destination_vendor_id': destination_vendor_id,
+                'destination_card_account_id': destination_card_account_id
+            }
+        )
+
+        return employee_mapping
```
Comment (ashwin1111) [inline fyle_accounting_mappings/models.py:586]: return employee_mapping, _
```suggestion
        return employee_mapping, _
```

File: fyle_accounting_mappings/serializers.py [hunk 3]
```diff
@@ -42,3 +54,80 @@ class MappingSerializer(serializers.ModelSerializer):
...
+            workspace_id=self.initial_data['workspace'],
+            attribute_type='EMPLOYEE'
+        ).first()
+
+        if not attribute:
+            raise serializers.ValidationError('No attribute found with this attribute id')
+        return source_employee
+
+    def validate_destination_employee(self, destination_employee):
+        if destination_employee and 'id' in destination_employee:
```
Comment (ashwin1111) [inline fyle_accounting_mappings/serializers.py:84]:
```suggestion
        if destination_employee and 'id' in destination_employee and destination_employee['id']:
```

File: fyle_accounting_mappings/serializers.py [hunk 4]
```diff
@@ -42,3 +54,80 @@ class MappingSerializer(serializers.ModelSerializer):
...
+                workspace_id=self.initial_data['workspace'],
+                attribute_type='EMPLOYEE'
+            ).first()
+
+            if not attribute:
+                raise serializers.ValidationError('No attribute found with this attribute id')
+        return destination_employee
+
+    def validate_destination_vendor(self, destination_vendor):
+        if destination_vendor and 'id' in destination_vendor:
```
Comment (ashwin1111) [inline fyle_accounting_mappings/serializers.py:96]:
```suggestion
        if destination_vendor and 'id' in destination_vendor and destination_vendor['id']:
```

File: fyle_accounting_mappings/serializers.py [hunk 5]
```diff
@@ -42,3 +54,80 @@ class MappingSerializer(serializers.ModelSerializer):
...
+                workspace_id=self.initial_data['workspace'],
+                attribute_type='VENDOR'
+            ).first()
+
+            if not attribute:
+                raise serializers.ValidationError('No attribute found with this attribute id')
+        return destination_vendor
+
+    def validate_destination_card_account(self, destination_card_account):
+        if destination_card_account and 'id' in destination_card_account:
```
Comment (ashwin1111) [inline fyle_accounting_mappings/serializers.py:108]:
```suggestion
        if destination_card_account and 'id' in destination_card_account and destination_card_account['id']:
```

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

**Pull request merged**